src/style.js

```diff
@@ -23,23 +14,32 @@ export default class JSXStyle extends Component {
       .join(' ')
   }
 
-  componentDidMount() {
-    styleSheetRegistry.add(this.props)
-  }
-
+  // probably faster than PureComponent (shallowEqual)
   shouldComponentUpdate(nextProps) {
-    return this.props.css !== nextProps.css
+    return (
+      this.props.styleId !== nextProps.styleId ||
+      // We do this check because `dynamic` is an array of strings or undefined.
+      // These are the computed values for dynamic styles.
+      String(this.props.dynamic) !== String(nextProps.dynamic)
+    )
   }
 
-  componentDidUpdate(prevProps) {
-    styleSheetRegistry.update(prevProps, this.props)
+  // Remove styles in advance.
```

Comment:
Can you explain what this means or what it's doing?

Reply:
> On update we then remove the old styles early enough aka on getSnapshotBeforeUpdate.

Reply:
this is the case when you have dynamic styles i.e. interpolations (tmpl literals expressions) that are based on props